tests/distances/test_scphylo_wrapper.py

```diff
@@ -63,3 +70,164 @@ def test_another_is_different_and_symmetric(similarity, model_tree) -> None:
         assert similarity.calculate(model_tree, root) == pytest.approx(
             similarity.calculate(root, model_tree)
         )
+
+
+def tree_gen(tree_type: str, seed: int) -> Callable[[int], TreeNode]:
+    """Return a tree."""
+    rng = jax.random.PRNGKey(seed)
+
+    def tree1_(n_nodes: int) -> TreeNode:
+        """Return a tree."""
+        if tree_type == "r":
+            return ti.generate_random_TreeNode(rng, n_nodes)
+        elif tree_type == "d":
+            return ti.generate_deep_TreeNode(rng, n_nodes)
+        else:  # tree_type == "s"
+            return ti.generate_star_TreeNode(n_nodes)
+
+    return tree1_
```

Comment:
Just playing around with currying ;)

Reply:
Nice! Here it's clear what it is for, but generally it's good to have precise docstrings what exactly the returned function is (as `Callable[[arg_types], output_type]` is often not enough to say what function is intended to do and what the arguments mean)

Reply:
OK understood - as it's just a test I didn't document too well. Thank you.